crates/ruma-macros/src/events/event_type.rs

```diff
@@ -161,6 +161,38 @@ fn generate_enum(
         })
         .collect::<syn::Result<_>>()?;
 
+    let from_ident_for_room = if ident == "StateEventType" || ident == "MessageLikeEventType" {
+        let match_arms: Vec<_> = deduped
+            .iter()
+            .map(|e| {
+                let v = e.to_variant()?;
+                let ident_var = v.match_arm(quote! { #ident });
+                let room_var = v.ctor(quote! { Self });
+
+                Ok(if e.has_type_fragment() {
+                    quote! { #ident_var (_s) => #room_var (_s) }
+                } else {
+                    quote! { #ident_var => #room_var }
+                })
+            })
+            .collect::<syn::Result<_>>()?;
+        let room_ident = Ident::new("RoomEventType", Span::call_site());
+
+        Some(quote! {
+            #[allow(deprecated)]
+            impl ::std::convert::From<#ident> for #room_ident {
```

Comment:
```suggestion


        Some(quote! {
            #[allow(deprecated)]
            impl ::std::convert::From<#ident> for #ruma_common::events::RoomEventType {
```

Reply:
According to the test it seems like it doesn't recognize `PrivOwnedStr` as the same type with this change.

Reply:
Oh... Maybe `crate::events` then? Since this is only supposed to be used in ruma-events.

Reply:
Or actually, since `RoomEventType` is generated by the same macro, it's fine for it to be just `RoomEventType`, but you don't need to put it in a separate `room_ident` variable first.